### DISCUSSION TITLE
Show and hide filters dynamically

### DISCUSSION THREAD

**Discussion by malsioufi** [appbaseio/reactivesearch#1663] [Q&A]
Hi, I am trying to build an interface using ReactiveSearch where the user can decide what filters to show and what to hide (similar functionality to "Add Filter" in Kibana interface).
Everything works as it should except for the URL. I have activated URLParams prop on all of my filters, and if I use a filter it updates the URL and the results section successfully, however, if I remove the filter from the interface, it updates the results but not the URL. 
At the same time, if I have a set of filters and removed them one by one, it only updates the URL when I remove the last one.
Do yo have any idea or suggestions to solve this issue?

**Comment by bietkul** [comment]
@malsioufi Ideally, it shouldn't happen because we update the URL when the component gets unmounted. Can you share a reproducible example(a codesandbox link would be better)?

**Reply by malsioufi** [reply]
Sure, you can find this example on the following link:
https://codesandbox.io/s/showandhidefilters-z601s
